src/plugins/organisation/PMRWindow/src/pmrwindowwindow.cpp

```diff
@@ -305,335 +214,34 @@ void PmrWindowWindow::on_filterValue_textChanged(const QString &pText)
-
-                                    std::sort(exposureFileNames.begin(), exposureFileNames.end(), sortExposureFiles);
-
-                                    mPmrWidget->addExposureFiles(exposureUrl, exposureFileNames);
-                                }
-                            }
-                        } else {
-                            hasExposureFileInformation = false;
-                        }
-                    } else {
-                        hasExposureFileInformation = false;
-                    }
-
-                    if (!hasExposureFileInformation)
-                        informationMessage = tr("No exposure file information could be found for <a href=\"%1\">%2</a>.");
-
-                    break;
-                }
-                }
-
-                // Check whether something wrong happened during the processing
-                // of our request
-
-                if (!informationMessage.isEmpty()) {
-                    QMessageBox::information( Core::mainWindow(), windowTitle(),
-                                              informationMessage.arg(exposureUrl, mExposureNames.value(exposureUrl))
-                                             +"<br/><br/>"+mInformationNoteMessage,
-                                              QMessageBox::Ok);
-                }
-            } else {
-                errorMessage = jsonParseError.errorString();
-            }
-        } else {
-            errorMessage = pNetworkReply->errorString();
-        }
-    } else {
-        internetConnectionAvailable = false;
-    }
-
-    // Some additional processing based on our PMR request
-
-    switch (pmrRequest) {
-    case ExposuresList:
-        // Ask our PMR widget to initialise itself
-
-        mPmrWidget->initialize(exposures, errorMessage,
-                               mGui->filterValue->text(),
-                               internetConnectionAvailable);
-
-        break;
-    case ExposureInformation:
-        // No additional processing needed
-
-        break;
-    case WorkspaceInformation:
-        // Clone the workspace, if possible and requested
-
-        if (   mWorkspaces.contains(exposureUrl)
-            && (Action(pNetworkReply->property(ActionProperty).toInt()) == CloneWorkspace)) {
-            doCloneWorkspace(mWorkspaces.value(exposureUrl));
-        }
-
-        break;
-    case ExposureFileInformation:
-        // Show the exposure files, if possible and requested
-
-        if (   !mNumberOfExposureFileUrlsLeft
-            &&  (Action(pNetworkReply->property(ActionProperty).toInt()) == ShowExposureFiles)) {
-            doShowExposureFiles(exposureUrl);
-        }
-
-        break;
-    }
-
-    // Show ourselves as not busy anymore
-
-    busy(false);
-
-    // Delete (later) the network reply
-
-    if (pNetworkReply)
-        pNetworkReply->deleteLater();
+    mExposures << PmrWindowExposure(pUrl, pName);
```

Comment:
Surely, this could be improved? I mean that right now, this slot is going to be called 576 times! Couldn't we retrieve all the exposures and then have them emitted? This also means that we wouldn't have to clear the list of exposures in ```PmrWindowWindow::on_refreshButton_clicked()```.

In fact, I don't understand what ```mExposures``` is doing in ```PmrWindowWindow```. If anything, it ought to be in ```PmrWindowWidget``` since this is where it is actually used. So, all ```PmrWindowWindow``` should do is pass on the list of exposures onto ```PmrWindowWidget```.